test/swim_test.js

```diff
@@ -191,14 +192,14 @@ test('suspicion subprotocol cannot be reenabled without all timers first being s
 });
 
 test('marks member faulty after suspect period', function t(assert) {
-    assert.plan(2);
+    assert.plan(1);
 
     var suspicion = createSuspicion();
     var member = suspicion.ringpop.membership.remoteMember;
     suspicion.setTimeout = function(fn) { return fn(); };
-    suspicion.ringpop.membership.update = function(changes) {
-        assert.equals(changes[0].address, member.address, 'updates correct member');
-        assert.equals(changes[0].status, 'faulty', 'marks member faulty');
+    // TODO Brittle test, just use real membership object
```

Comment:
Generally when writing tests, i tend to not mock out any logic, even if it's a separate unit.

This does make your tests a hybrid unit/integration test instead of a pure unit test, but most of the times you'd rather test all the logic and avoid having to write the unit twice, once in the code and once in the mock.

I tend to only mock out IO or thin clients around IO. `tchannel` would be a good thing to mock out as it's a thin client around IO.